### DISCUSSION TITLE
how to make changes in YOLOv8 backbone with resnet50

### DISCUSSION THREAD

**Discussion by suciptr** [ultralytics/ultralytics#18194] [Q&A]
hello.
I am an student with very limited knowledge of Deep Learning and Artificial Neural Networks. 
I have an assignment to create a classification model with yolov8, then I want to make some modifications to the model that I have trained, I want to change the yolov8 backbone to resnet50.
Can someone teach and guide me? I will be very grateful.

**Comment by pderrenger** [comment]
👋 Hello @suciptr, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) as a starting point for exploring YOLOv8, which includes extensive [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that might help you better understand how to use and customize the model.

If this is a custom training ❓ question, it’s helpful to provide as much context as possible, such as your dataset samples, current approach, and any relevant logs or code snippets. This will assist Ultralytics engineers and the community in providing focused advice. For your specific goal of modifying the YOLOv8 backbone to ResNet50, this may involve deeper customization of the model architecture.

For real-time discussions and peer support, you can join our [Discord](https://discord.com/invite/ultralytics) 🎧. You may also find detailed discussions and shared knowledge on our [Discourse platform](https://community.ultralytics.com) or [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Before proceeding, ensure you have the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) installed in a [**Python>=3.8**](https://www.python.org/) environment. Up-to-date dependencies avoid many common issues:

```bash
pip install -U ultralytics
```

## Notebooks for Testing

You can experiment with YOLOv8 in interactive cloud environments with free GPUs:

- <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a>
- <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>
- <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

## Status

You can check the status of our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests below. A green badge indicates that all automated tests are currently passing:

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This is an automated response to guide you toward available resources and community forums 😊. Rest assured, an Ultralytics engineer will review your question and assist you soon!

**Reply by suciptr** [reply]
I'm a beginner, I don't understand at all what I should do first. Can you help me do it?

**Reply by glenn-jocher** [reply]
@suciptr hi! To help you modify YOLOv8's backbone to ResNet50, I recommend starting with the basic YOLOv8 training example first to get familiar with the framework. Here's a simple approach:

```python
from ultralytics import YOLO

# Load a base YOLOv8 model
model = YOLO('yolov8n.yaml')  

# Train it on your dataset
model.train(data='path/to/data.yaml', epochs=100)
```

Once you're comfortable with basic training, you can explore modifying the model architecture by editing the YAML configuration file to use ResNet50 as the backbone. Would you like me to provide more specific guidance on working with the YAML config?

Let me know if you need help with any particular step and I'll be happy to assist further.

**Reply by suciptr** [reply]
yes please.
I have run the program above with my dataset, producing MAP 90s, because the results are good I was asked to change the backbone using resnet50. 
I have tried several times to change the program code but still can't, because I have no basic in this matter. so please help me, I would be very helped 💫

**Reply by glenn-jocher** [reply]
To change the YOLOv8 backbone to ResNet50, you'll need to modify the model's YAML configuration file to define ResNet50 as the backbone. Unfortunately, YOLOv8 does not natively support ResNet50, so you'll need to integrate it manually by customizing the model architecture. This requires defining the ResNet50 backbone in the YAML file and ensuring compatibility with the YOLO head layers.

If you're new to these concepts, I recommend thoroughly reviewing the PyTorch ResNet50 implementation and Ultralytics' model configuration files for guidance. You can also explore the [Ultralytics Docs](https://docs.ultralytics.com/) or raise detailed questions in the community forums for additional support.

**Reply by suciptr** [reply]
is there any file have to change beside model.yaml?

**Reply by glenn-jocher** [reply]
To modify YOLOv8's backbone to ResNet50, you need to edit the `model.yaml` file to define the ResNet50 architecture. Additionally, you may need to adjust the `ultralytics/nn/tasks.py` file to ensure the ResNet50 model is correctly parsed and integrated. Let me know if you need further clarification.
